### PR TITLE
refactor: keychain

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/5462301083).

Precursory refactor, moving around keychain logic

### CODE REVIEW

File: src/app/query/bitcoin/ordinals/use-next-fresh-taproot-address.query.ts [hunk 1]
```diff
@@ -14,20 +14,20 @@ import { useCurrentNetwork } from '@app/store/networks/networks.selectors';
...
     ['next-taproot-address', currentAccountIndex, network.id] as const,
     async () => {
-      if (!keychain) throw new Error('Expected keychain to be provided');
+      if (!account) throw new Error('Expected keychain to be provided');
 
       async function taprootAddressIndexActivity(index: number) {
         const address = getTaprootAddress({
           index,
-          keychain,
+          keychain: account?.keychain,
```
Comment (alter-eggo) [inline src/app/query/bitcoin/ordinals/use-next-fresh-taproot-address.query.ts:30]: no need in ?.

File: src/app/query/bitcoin/ordinals/use-taproot-address-utxos.query.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export function useTaprootAccountUtxosQuery() {
       ) {
         const address = getTaprootAddress({
           index: addressIndexCounter.getValue(),
-          keychain,
+          keychain: account?.keychain,
```
Comment (alter-eggo) [inline src/app/query/bitcoin/ordinals/use-taproot-address-utxos.query.ts:41]: hm, wonder why we do `account` check everywhere, and here not

**Review by alter-eggo — APPROVED** [review]

File: src/app/query/bitcoin/ordinals/use-taproot-address-utxos.query.ts (hunk 2 above)
Comment (kyranjamie) [reply src/app/query/bitcoin/ordinals/use-taproot-address-utxos.query.ts:41]: In both cases we have to check for `undefined`, not exactly sure why one throws and the other does

Comment (kyranjamie) [reply src/app/query/bitcoin/ordinals/use-taproot-address-utxos.query.ts:41]: @alter-eggo making further changes to these in other PR, going to merge, but hope for feedback there too

**Pull request merged**